core/state/statedb.go

```diff
@@ -842,10 +842,6 @@ func (s *StateDB) Commit(deleteEmptyObjects bool) (common.Hash, error) {
 		if account.Root != emptyRoot {
 			s.db.TrieDB().Reference(account.Root, parent)
 		}
-		code := common.BytesToHash(account.CodeHash)
-		if code != emptyCode {
-			s.db.TrieDB().Reference(code, parent)
-		}
```

Comment:
Did you deliberately remove the code from GC? If yes, shouldn't we skip `triedb.InsertCode` altogether and just drop it into the database?

Reply:
Yes, contract code garbage collection only makes sense if the code can be changed but it's not allowed in the protocol level? So for all codes we can blindly insert them.

> shouldn't we skip triedb.InsertCode altogether 

Let me see how to do it.